uefi/src/table/runtime.rs

```diff
@@ -371,19 +371,66 @@ pub struct TimeParams {
     pub daylight: Daylight,
 }
 
-/// Error returned by [`Time`] methods if the input is outside the valid range.
-#[derive(Copy, Clone, Debug, Default, Eq, PartialEq)]
-pub struct TimeError;
+/// Error returned by [`Time`] methods. A bool value of `true` means
+/// the specified field is outside of its valid range.
+#[allow(missing_docs)]
+#[derive(Debug, Default, PartialEq, Eq)]
+pub struct TimeError {
+    pub year: bool,
+    pub month: bool,
+    pub day: bool,
+    pub hour: bool,
+    pub minute: bool,
+    pub second: bool,
+    pub nanosecond: bool,
+    pub timezone: bool,
+    pub daylight: bool,
+}
+
+#[cfg(feature = "unstable")]
+impl core::error::Error for TimeError {}
 
 impl Display for TimeError {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        write!(f, "{self:?}")
+        if *self == TimeError::default() {
+            write!(f, "No erroneous fields found in EFI Time.")?;
+        } else {
+            write!(f, "Erroneous fields found in EFI Time:")?;
```

Comment:
very nit: if you use `writeln!` here, you do not have to repeat `\n` so often 😁

Reply:
i was being very nitty myself- i wrote it this way so that the last line doesn't end with `\n` but I agree that `writeln!` looks cleaner